api/app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,109 @@
+<?php namespace App\Http\Controllers;
+
+use RuntimeException;
+use App\Models\AuthToken;
+use Tymon\JWTAuth\JWTAuth;
+use Illuminate\Http\Request;
+use App\Exceptions\UnauthorizedException;
+use Tymon\JWTAuth\Exceptions\JWTException;
+use Tymon\JWTAuth\Exceptions\TokenExpiredException;
+
+class AuthController extends BaseController
+{
+    /**
+     * JWT Auth
+     *
+     * @var Tymon\JWTAuth\JWTAuth
+     */
+    protected $jwtAuth;
+
+    /**
+     * Assign dependencies.
+     *
+     * @param  JWTAuth  $jwtAuth
+     * @return void
+     */
+    public function __construct(JWTAuth $jwtAuth)
+    {
+        $this->jwtAuth = $jwtAuth;
+    }
+
+    /**
+     * Get a login token.
+     *
+     * @param  Request  $request
+     * @return Response
+     */
+    public function login(Request $request)
+    {
+        $credentials = [
+            'email' => $request->getUser(),
+            'password' => $request->getPassword()
+        ];
+
+        try {
+            if (!$token = $this->jwtAuth->attempt($credentials)) {
+
+                throw new UnauthorizedException;
+            }
+        } catch (JWTException $e) {
+            throw new RuntimeException('Token could not be encoded.');
```

Comment:
In what scenario can this happen? Is runtime an appropriate exception? Either way the exception should have `$e` added as previous